int32 control_max_time
int32 control_type
